### ISSUE TITLE
handleKeyDown: Tab not triggering submit

### ISSUE THREAD

**Issue by cgosborne** [trevoreyre/autocomplete#23]
When using the VueComponent layer, when the AutocompleteCore handles the "Tab" keypress, it doesn't trigger the onSubmit method. The value gets filled to the input but I can't update my object when the submit event is triggered. I'm not sure if this was by design or not, but I would expect Tab and Enter to be handled similarly.

**Comment by trevoreyre** [comment]
Yes, this is by design. I know a lot of autocomplete implementations handle this both ways, and I went back and forth a number of times about whether tab should also fire the submit event. Ultimately, I tried to follow the [WAI-ARIA Authoring Practices guide](https://www.w3.org/TR/wai-aria-practices-1.1/examples/combobox/aria1.1pattern/listbox-combo.html) as closely as I could, and they only mention  accepting input on the enter key, so I left it out.

If you do need the tab functionality, I believe you could add your own `keypress` handler to detect the tab key and call whatever your submit function is manually. Not quite as clean I know, but it should be doable.

**Comment by trevoreyre** [comment]
@cgosborne I'm going to go ahead and close this. Feel free to re-open if you have additional questions or concerns.

**Comment by nextmat** [comment]
I'm running into this as well. 

Wiring up a custom handler is pretty straightforward if the value you want submitted is the same as what you are displaying. However if you are dealing with objects as your results and displaying a sub-property like `name`, there doesn't seem to be an easy way to retrieve the state of what just got selected?

It would be great if this was either configurable or if there was an easier way to ask the component what the selected value was, as discussed in #44 .

**Comment by trevoreyre** [comment]
@nextmat Agreed. I'll re-open and add it to the list of use cases to check in v3.

**Comment by nextmat** [comment]
Sounds great!

In case it helps anyone in the meantime, I was able to kludge together something that works (but feels pretty brittle) through reaching into the data properties of the component. My parent component's handler ended up looking like this:

```js
handleTab() {
  const results = this.$refs.autocomplete.results;
  const index = this.$refs.autocomplete.selectedIndex;
  if (results[index]) this.selectResult(results[index]);
}
```

**Comment by onecreative** [comment]
I'm not using the Vue version, just js. But this is causing some issues for me as well.

Only accepting input on Enter makes sense until you have a feature called `autoSelect`. Seems to me that when that option is set to true, `onSubmit` should fire after tabbing out of the field.

**Comment by guastallaigor** [comment]
I was having the same problems, but I kind of did it differently in a simpler way.

I add a blur event listener (`@blur="handleBlur"`) to this Autocomplete component, instead of key events.

I'm also saving all the filtered items in another data property. Then, when the user leaves the autocomplete, the blur is triggered, and I get the first item from all the items array, so I call the handleSubmit (same way as the example).

To me this works, the `auto-select` prop is set to `true`, so the user always going to select the first item from the array.

```js
search(input) {
  if (input.length < 1) {
    return [];
  }

  this.itemsFiltered = this.items.filter(item => {
    return item.name.toLowerCase().startsWith(input.toLowerCase());
  });

  return this.itemsFiltered;
},
handleBlur() {
  const results = this.itemsFiltered;
  const index = 0;
  this.handleSubmit(results[index]);
}
```

**Comment by jelicich** [comment]
@nextmat Thanks for providing that solution. I'm facing the same scenario, however, for some reason on the `keydown.tab` the `results` and `selectedIndex` are already empty (being results a length 0 array and the selected index -1).
Any ideas on why the tab event might be fired after the autocomplete is reset? 
Weird thing is that when adding a breakpoint on my tab callback I can still see the results list rendered.

**Comment by nextmat** [comment]
@jelicich - it has been a long time since l looked at this so unfortunately I don't have a lot of context anymore. This was working with the version of Vue I was using at the time but it has been a while so the event sequencing might work differently now?
